src/app/src/main/kotlin/org/vaccineimpact/api/app/OneTimeLink.kt

```diff
@@ -47,11 +51,26 @@ data class OneTimeLink(val action: OneTimeAction, val payload: Map<String, Strin
             val rawAction = claims["action"].toString()
             val action = Deserializer().parseEnum<OneTimeAction>(rawAction)
             val rawPayload = claims["payload"].toString()
+            val rawQueryParams = claims["query"]?.toString()
             val payload = rawPayload
                     .split('&')
                     .map { it.split('=') }
                     .associateBy({ it[0] }, { it[1] })
-            return OneTimeLink(action, payload)
+
+            val queryParams =
+                    if (rawQueryParams.isNullOrEmpty())
+                    {
+                        mapOf()
+                    }
+                    else
+                    {
+                        rawQueryParams!!
```

Comment:
If you replace the if statement with `if (rawQueryParams == null || rawQueryParams == "")` then you won't need the `!!` null coercion.